services/src/handlers/workflows.rs

```diff
@@ -379,70 +382,99 @@ pub struct RasterDatasetFromWorkflowResult {
     upload: UploadId,
 }
 
-/// Create a new dataset from the result of the workflow given by its `id` and the dataset parameters in the request body.
-/// Returns the id of the created dataset and upload
-#[utoipa::path(
-    tag = "Workflows",
-    post,
-    path = "/datasetFromWorkflow/{id}",
-    request_body = RasterDatasetFromWorkflow,
-    responses(
-        (status = 200, description = "Id of created dataset and upload", body = RasterDatasetFromWorkflowResult,
-            example = json!({"upload": "3086f494-d5a4-4b51-a14b-3b29f8bf7bb0", "dataset": {"type": "internal", "datasetId": "94230f0b-4e8a-4cba-9adc-3ace837fe5d4"}})
-        )
-    ),
-    params(
-        ("id" = WorkflowId, description = "Workflow id")
-    ),
-    security(
-        ("session_token" = [])
-    )
-)]
-async fn dataset_from_workflow_handler<C: Context>(
+impl TaskStatusInfo for RasterDatasetFromWorkflowResult {}
+
+async fn create_dataset<C: Context>(
+    info: RasterDatasetFromWorkflow,
+    mut slice_info: Vec<GdalLoadingInfoTemporalSlice>,
+    result_descriptor: &geoengine_operators::engine::RasterResultDescriptor,
+    ctx: &C,
+    session: <C as Context>::Session,
+) -> Result<DatasetId> {
+    let dataset_id = DatasetId::new();
+    //TODO: Recognize MetaDataDefinition::GdalMetaDataRegular
+    let meta_data = if slice_info.len() == 1 {
+        let loading_info_slice = slice_info.pop().expect("slice_info has len one");
+        let time = Some(loading_info_slice.time);
+        let params = loading_info_slice
+            .params
+            .expect("params is always set in raster_stream_to_geotiff"); //TODO: What is the use-case for None in the GdalMetaDataList?
```

Comment:
be26049d69aa5e5b30df269083c554ad1763b180

I changed the expect message, but the expect should be valid for this case, because a static dataset should not consist of just a gap.